crates/wasm-mutate/src/mutators/peephole/rules.rs

```diff
@@ -415,6 +415,16 @@ impl PeepholeMutator {
                     "?x" => "i64.const.1"
                         if self.is_type("?x", PrimitiveTypeInfo::I64)
                 ),
+                rewrite!(
+                    "replace-with-f32-1";
+                    "?x" => "f32.const.1"
+                        if self.is_type("?x", PrimitiveTypeInfo::F32)
+                ),
+                rewrite!(
+                    "replace-with-f64-1";
+                    "?x" => "f64.const.1"
+                        if self.is_type("?x", PrimitiveTypeInfo::F64)
```

Comment:
I think this may not work as intended since this produces a float equivalent to `f64::from_bits(1)`, so for this to actually use `1.0` I think it'd need to modify the AST to store a `f64` or a newtype wrapper or something like that.

Reply:
I… went with hardcoding the bit pattern for 1.0 instead. Not the prettiest solution, but it isn't clear to me if we'll be interested in any other constants and the rewrite rule name acts as a pretty good documentation as to what is happening here.